test/ng/compileSpec.js

```diff
@@ -12111,6 +12141,626 @@ describe('$compile', function() {
+          //Test regex to make sure doesn't mistake parts of url for width descriptors
+          'http://example.com/image1.png?w=48w,http://example.com/image2.png 64w':'http://example.com/image1.png?w=48w,http://example.com/image2.png 64w',
+          'http://example.com/image1.png 1x,http://example.com/image2.png 64w':'http://example.com/image1.png 1x,http://example.com/image2.png 64w',
+          'http://example.com/image1.png,http://example.com/image2.png':'http://example.com/image1.png ,http://example.com/image2.png',
+          'http://example.com/image1.png ,http://example.com/image2.png':'http://example.com/image1.png ,http://example.com/image2.png',
+          'http://example.com/image1.png, http://example.com/image2.png':'http://example.com/image1.png ,http://example.com/image2.png',
+          'http://example.com/image1.png , http://example.com/image2.png':'http://example.com/image1.png ,http://example.com/image2.png',
+          'http://example.com/image1.png 1x, http://example.com/image2.png 2x, http://example.com/image3.png 3x':
+            'http://example.com/image1.png 1x,http://example.com/image2.png 2x,http://example.com/image3.png 3x',
+          'javascript:doEvilStuff() 2x': 'unsafe:javascript:doEvilStuff() 2x',
+          'http://example.com/image1.png 1x,javascript:doEvilStuff() 2x':'http://example.com/image1.png 1x,unsafe:javascript:doEvilStuff() 2x',
+          'http://example.com/image1.jpg?x=a,b 1x,http://example.com/ima,ge2.jpg 2x':'http://example.com/image1.jpg?x=a,b 1x,http://example.com/ima,ge2.jpg 2x',
+          //Test regex to make sure doesn't mistake parts of url for pixel density descriptors
+          'http://example.com/image1.jpg?x=a2x,b 1x,http://example.com/ima,ge2.jpg 2x':'http://example.com/image1.jpg?x=a2x,b 1x,http://example.com/ima,ge2.jpg 2x'
+        };
+
+        forEach(testSet, function(ref, url) {
+          $rootScope.testUrl = url;
+          $rootScope.$digest();
+          expect(element.prop('srcset')).toEqual(ref);
+        });
+      }));
+    });
+
+    describe('a[href] sanitization', function() {
+      it('should NOT require trusted values for whitelisted values', inject(function($rootScope, $compile) {
+        $rootScope.testUrl = 'http://example.com/image.png'; // `http` is whitelisted
+        element = $compile('<a ng-prop-href="testUrl"></a>')($rootScope);
+        $rootScope.$digest();
+        expect(element.prop('href')).toEqual('http://example.com/image.png');
+
+        element = $compile('<a ng-prop-href="testUrl"></a>')($rootScope);
+        $rootScope.$digest();
+        expect(element.prop('href')).toEqual('http://example.com/image.png');
+      }));
+
+      it('should accept trusted values for non-whitelisted values', inject(function($rootScope, $compile, $sce) {
+        $rootScope.testUrl = $sce.trustAsUrl('javascript:foo()'); // `javascript` is not whitelisted
```

Comment:
Similar to `img[src]` we should check the different trust levels. E.g. ResourceUrl and Url should pass, but in this case MediaUrl should fail, right?

Reply:
I'm really not sure which should pass/fail. I'll need some help with those tests (which are still a TODO).